src/contributors/MetricsTableContributor.ts

```diff
@@ -0,0 +1,425 @@
+    }
+
+    /** @override */
+    public fetchData(collaborationEvent: CollaborationEvent): Observable<ComputableResponse> {
+        if (collaborationEvent.id !== this.identifier || collaborationEvent.operation === OperationEnum.remove) {
+            const allKeys = new Set<string>();
+            /** Base aggregations to get terms and their metrics for each collection (vector) */
+            return forkJoin(this.table.vectors.map(v =>
+                this.collaborativeSearcheService.resolveButNotAggregation([projType.aggregate, [v.getAggregation()]],
+                    this.collaborativeSearcheService.collaborations,
+                    v.collection,
+                    this.identifier, {}, false, this.cacheDuration
+                ).pipe(
+                    map(ar => {
+                        const keys = new Set(ar.elements.map(e => e.key));
+                        keys.forEach(k => allKeys.add(k));
+                        return ({
+                            collection: v.collection,
+                            aggregationResponse: ar,
+                            keys,
+                            missingKeys: new Set<string>(),
+                            vector: v
+                        });
+                    })
+                )
+            )).pipe(
+                /** For each vector, recuperate the missing keys that exist in other vectors */
+                map(metricsResponses => {
+                    metricsResponses.forEach(mr => {
+                        allKeys.forEach(k => {
+                            if (!mr.keys.has(k)) {
+                                mr.missingKeys.add(k);
+                            }
+                        });
+                    });
+                    return metricsResponses;
+                }),
+                /** The following block launches a list of Complementary aggregations for each vector (only in case of missing keys),
+                 *  in order to fetch the missing keys and complete the table. */
+                mergeMap(metricsResponses => forkJoin(metricsResponses.map(mr => {
+                    if (mr.missingKeys.size === 0) {
+                        return of(mr);
+                    } else {
+                        const termsToInclude = Array.from(mr.missingKeys);
+                        return this.collaborativeSearcheService
+                            .resolveButNotAggregation([projType.aggregate, [mr.vector.getAggregation(termsToInclude)]],
+                                this.collaborativeSearcheService.collaborations,
+                                mr.collection,
+                                this.identifier, {}, false, this.cacheDuration
+                            ).pipe(
+                                map(ar => {
+                                    const keys = new Set(ar.elements.map(e => e.key));
+                                    keys.forEach(k => {
+                                        allKeys.add(k);
```

Comment:
The key should already be in allKeys so this line is not useful

Reply:
I agree